src/modules/Game.class.js

```diff
@@ -50,19 +158,97 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.#addNumber();
+    this.#addNumber();
+    this.#status = 'playing';
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.#applyInitialState();
+    this.#score = 0;
+    this.#status = 'idle';
+  }
+
+  #applyInitialState() {
+    this.#state = this.#initialState.map((row) => [...row]);
+  }
+  /**
+   * Checks if any move is possible
+   * @returns {boolean}
+   */
+  #isMovePossible() {
+    const gameCopy = new Game(this.getState(), true);
+
+    return (
+      gameCopy.moveLeft() ||
+      gameCopy.moveRight() ||
+      gameCopy.moveUp() ||
+      gameCopy.moveDown()
+    );
+  }
+  /**
+   * Returns if adding number to state was succesfull
+   * @returns {boolean}
+   */
+  #addNumber() {
+    const number = Math.random() > 0.1 ? 2 : 4;
```

Comment:
The probability of adding a '2' or '4' is hardcoded. Consider making these values and probabilities configurable to allow for easy adjustments and potentially different game modes.